provider/identity_provider_resource.go

```diff
@@ -103,28 +110,138 @@ func (t *IdentityProviderResourceType) gitlabSchema() tfsdk.NestedAttributes {
 	return tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
 		"ca": {
 			Description: "Optional trusted certificate authority bundle.",
-			Type:     types.StringType,
-			Optional: true,
+			Type:        types.StringType,
+			Optional:    true,
 		},
 		"client_id": {
 			Description: "Client identifier of a registered Gitlab OAuth application.",
-			Type:     types.StringType,
-			Required: true,
+			Type:        types.StringType,
+			Required:    true,
 		},
 		"client_secret": {
 			Description: "Client secret issued by Gitlab.",
-			Type:      types.StringType,
-			Required:  true,
-			Sensitive: true,
+			Type:        types.StringType,
+			Required:    true,
+			Sensitive:   true,
 		},
 		"url": {
 			Description: "URL of the Gitlab instance.",
-			Type:     types.StringType,
-			Required: true,
+			Type:        types.StringType,
+			Required:    true,
+		},
+	})
+}
+
+func (t *IdentityProviderResourceType) githubSchema() tfsdk.NestedAttributes {
```

Comment:
this method need to be outside from IdentityProviderResourceType struct

Reply:
Followed the same structure as the others - but you are right. Will be moved to the new `idps` package